cds_ils/importer/api.py

```diff
@@ -43,61 +42,68 @@ def process_dump(data, provider, mode, source_type):
-    if eager:
-        return process_dump(data, provider, mode, source_type=source_type)
-    else:
-        process_dump.delay(data, provider, mode, source_type=source_type)
 
 
-def import_from_xml(log_id, source_path, source_type, provider, mode):
+def import_from_xml(log, source_path, source_type, provider, mode,
+                    eager=False):
     """Load a single xml file."""
-    log = ImporterTaskLog.query.filter_by(id=log_id).first()
-    entry_data = None
     try:
-        with open(source_path) as source:
+        validate_import(provider, mode, source_type)
+
+        with open(source_path, "r") as source:
             records_list = list(get_records_list(source))
 
             # update the entries count now that we know it
-            log.entries_count = len(records_list)
-            db.session.commit()
-
-            for i, record in enumerate(records_list):
-                entry_data = dict(
-                    import_id=log.id,
-                    entry_index=i,
-                )
+            log.set_entries_count(records_list)
+            for record in records_list:
+                record_recid = get_record_recid_from_xml(record)
+
+                click.echo(f"Processing record {record_recid}")
                 try:
                     report = import_record(record, provider, mode,
                                            source_type=source_type,
-                                           eager=True)
+                                           eager=eager)
+                    ImportRecordLog.create_success(
+                        log.id, record_recid, report)
                 except (LossyConversion, RecordNotDeletable,
-                        ProviderNotAllowedDeletion) as e:
-                    ImporterTaskEntry.create_failure(entry_data, e)
+                        ProviderNotAllowedDeletion, SeriesImportError) as e:
+                    ImportRecordLog.create_failure(log.id, record_recid,
+                                                   str(e.message))
                     continue
                 except IlsValidationError as e:
-                    records_logger.error(
-                        "@FILE TASK: {0} FATAL: {1}".format(
-                            log_id,
+                    click.echo(
+                        "TASK: {0} FATAL: {1}".format(
+                            log.id,
                             str(e.original_exception.message),
                         )
                     )
-                    ImporterTaskEntry.create_failure(entry_data, e)
+                    ImportRecordLog.create_failure(
+                        log.id, record_recid,
+                        str(e.original_exception.message)
+                    )
                     continue
-
-                ImporterTaskEntry.create_success(entry_data, report)
     except Exception as e:
-        records_logger.error(
-            "@FILE TASK: {0} ERROR: {1}".format(log_id, str(e))
+        click.echo(
```

Comment:
As discussed in the past, personally I don't love to use the `click.echo` where you are not sure that it will be only called by a CLI command. An API file might be called by other tasks I guess. Why remove the logger and use the click.echo?